lib/ABI/FunctionType.cpp

```diff
@@ -52,74 +55,95 @@ bool verify(const SortedVector<RegisterType> &UsedRegisters,
   return true;
 }
 
-constexpr static model::PrimitiveTypeKind::Values
-selectTypeKind(model::Register::Values) {
-  // TODO: implement a way to determine the register type. At the very least
-  // we should be able to differentiate GPRs from the vector registers.
-
-  return model::PrimitiveTypeKind::PointerOrNumber;
+/// \brief A helper used to differentiate vector registers.
+///
+/// \param Register Any CPU register the model is aware of.
+///
+/// \return `true` if `Register` is a vector register, `false` otherwise.
+constexpr bool isVectorRegisterHelper(model::Register::Values Register) {
+  namespace PrimitiveTypeKind = model::PrimitiveTypeKind;
+  return model::Register::primitiveKind(Register) == PrimitiveTypeKind::Float;
 }
 
-static model::QualifiedType
-buildType(model::Register::Values Register, model::Binary &TheBinary) {
-  model::PrimitiveTypeKind::Values Kind = selectTypeKind(Register);
-  size_t Size = model::Register::getSize(Register);
-  return model::QualifiedType(TheBinary.getPrimitiveType(Kind, Size), {});
-}
-
-static model::QualifiedType
-buildGenericType(model::Register::Values Register, model::Binary &TheBinary) {
-  constexpr auto Kind = model::PrimitiveTypeKind::Generic;
-  size_t Size = model::Register::getSize(Register);
-  return model::QualifiedType(TheBinary.getPrimitiveType(Kind, Size), {});
-}
+/// \brief Helps detecting unsupported ABI trait definition with respect to
+///        the way they return the return values.
+///
+/// \note Make this function `consteval` after clang-14 is available.
+///
+/// \tparam ABI the ABI, `abi::Trait` of which is to be checked.
+///
+/// \return `true` if the ABI is valid, `false` otherwise.
+template<model::ABI::Values ABI>
+static constexpr bool verifyReturnValueLocationHelper() {
```

Comment:
* I think `static` is not necessary for `template` functions, isn't it?
* No need for the helper suffix.

Reply:
1. Ah, whoops, this one used to take ABI as an argument. I forgot to remove `static`, my bad.
2. 👌

Reply:
Both dropped.